diff -urN orig/CastSink.cpp new/CastSink.cpp
--- orig/CastSink.cpp	Thu Feb 28 11:49:25 2002
+++ new/CastSink.cpp	Thu Mar 28 18:40:55 2002
@@ -5,7 +5,7 @@
    Tyrell DarkIce
 
    File     : CastSink.cpp
-   Version  : $Revision: 1.7 $
+   Version  : $Revision: 1.8 $
    Author   : $Author: darkeye $
    Location : $Source: /cvsroot/darkice/darkice/src/CastSink.cpp,v $
    
@@ -42,7 +42,7 @@
 /*------------------------------------------------------------------------------
  *  File identity
  *----------------------------------------------------------------------------*/
-static const char fileid[] = "$Id: CastSink.cpp,v 1.7 2002/02/28 09:49:25 darkeye Exp $";
+static const char fileid[] = "$Id: CastSink.cpp,v 1.8 2002/03/28 16:40:55 darkeye Exp $";
 
 
 /* ===============================================  local function prototypes */
@@ -75,10 +75,12 @@
     this->isPublic       = isPublic;
     this->bufferDuration = bufferDuration;
 
-    bufferedSink = socket ? new BufferedSink( socket,
-                                     (bitRate * 1024 / 8) * bufferDuration)
-                          : 0;
+    int       bufferSize = bitRate ? (bitRate * 1024 / 8) * bufferDuration
+                                   : (128 * 1024 / 8) * bufferDuration;
 
+    bufferedSink = socket ?  new BufferedSink( socket,
+                                     (bufferSize * 1024 / 8) * bufferDuration)
+                          : 0;
 }
 
 
@@ -144,6 +146,10 @@
   $Source: /cvsroot/darkice/darkice/src/CastSink.cpp,v $
 
   $Log: CastSink.cpp,v $
+  Revision 1.8  2002/03/28 16:40:55  darkeye
+  slight changes to allow for variable bitrate streams
+  (where the value of bitrate is 0)
+
   Revision 1.7  2002/02/28 09:49:25  darkeye
   added possibility to save the encoded stream to a local file only
   (no streaming server needed)
